database/download.go

```diff
@@ -52,48 +61,84 @@ func (db *DB) DownloadCreate(ctx context.Context, user User, skylink Skylink) (*
 }
 
 // DownloadsBySkylink fetches all downloads of this skylink
-func (db *DB) DownloadsBySkylink(ctx context.Context, skylink Skylink, offset, limit int) ([]Download, error) {
+func (db *DB) DownloadsBySkylink(ctx context.Context, skylink Skylink, offset, limit int) ([]DownloadResponseDTO, error) {
```

Comment:
This is not usually how list functions work, how do you page with this? Usually what you return is an object that contains `items` and `count`. Most mongo drivers and libraries have a `findAndCount` for this reason. Allowing you to query using an `offset` and `limit` and still get the overall count.

Reply:
I changed the response to look like this:
```json
{
    "items": [
        {
            "id": "60227e000fb1b80cf4f795f4",
            "skylink": "HELLO__5MJyl1DKyfbuQMY_XBOBC9bR7idiU6isp6LXxEw",
            "name": "This is my HELLO skyfile!",
            "size": 132412341234,
            "downloadedOn": "2021-02-09T12:20:16.869Z"
        }
    ],
    "offset": 1,
    "pageSize": 1,
    "totalCount": 7
}
```

Unfortunately, I couldn't manage to do it within the same aggregation pipeline. I'm not sure if it's even possible or desirable. I'm doing it with a separate aggregation call that just filters the results by the same `matchStage` criteria and then counts them.